### PR TITLE
This commit leads to Angular2.0.0 completely.

### PR DESCRIPTION
**Pull request by devcken**
- Introduce peerDependencies in package.json.
- Arrange several dependencies.
- LocalStorageEmitter#LocalStorageSubscriber signature modified.
  from LocalStorageSubscriber(appPromise: Promise<ComponentRef<any>>) to LocalStorageSubscriber(promise: Promise<NgModuleRef<any>>)
- In WebStrage, chages type of variables implicitly declared with any. This provides type-safe codes.

### CODE REVIEW

**Comment by ross-nordstrom** [comment]
Any update on this? I see that you committed to master @devcken but I don't see a corresponding NPM release

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,212 @@
+### Eclipse template
```
Comment (marcj) [inline .gitignore:1]: can you please remove that file

**Comment by griga** [comment]
until there where correct npm version you can always use

`npm install https://github.com/devcken/angular2-localstorage --save`

**Comment by devcken** [comment]
Apologize for my late.
After I pulled this request, I have just read conversations.

angular2-localstorage have a problem which the emitter doesn't work. :( So, it has lost the power of change detection. Maybe you guys will need to detect the changes in local(or session) storage.

I will try to fix it this weekend.

Hi, @marcj Do you mean that remove whole changes in .gitignore or just that line? I have putted some commons(OS, IDE, stuff related to language,...) in it.

**Comment by devcken** [comment]
Finally, I have just modified totally.
In this project, I didn't write some examples, but I tested on my another project.
I think it works fine.

File: .gitignore (hunk 1 above)
Comment (marcj) [reply .gitignore:1]: The whole gitignore file please.

Comment (devcken) [reply .gitignore:1]: What do you mean the "whole" gitignore file? I'm sorry that I don't understand.
It has 212 line now. It still include "node_modules", "typings". I just added another template for OS, IDE, etc.

**Comment by AilisObrian** [comment]
```
ERROR in [default] /Users/obrian/projects/____/____/node_modules/angular2-localstorage/dist/decorator/webstorage.d.ts:1:0
Cannot find type definition file for 'core-js'.
```
In my case, using this patchset needs to modify my package.json for adding @types/core-js.

**Comment by AilisObrian** [comment]
Applying this patchset, needs to add @types/core-js dependency.
but It crash all my angular-cli-project.

angular/angular-cli#1908

**Comment by AhsanAyaz** [comment]
Guys. Any plans for merging this pull request?

**Comment by marcj** [comment]
Merged for the moment, please test guys and send pull requests if you find something. @devcken, very good job, thanks! 👍

**Pull request merged**